src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -271,41 +273,83 @@ export class IntegrationListDirectory extends AsyncComponent<
     );
   }, TEXT_SEARCH_ANALYTICS_DEBOUNCE_IN_MS);
 
+  /**
+   * Get filter parameters and guard against `queryString.parse` returning arrays.
+   */
+  getFilterParameters = (): {searchInput: string; selectedCategory: string} => {
+    const {category, search} = queryString.parse(this.props.location.search);
+
+    const selectedCategory = Array.isArray(category) ? category[0] : category || '';
```

Comment:
@mgaeta I am confused when these would be arrays

Reply:
The object returned by [queryString.parse](https://nodejs.org/api/querystring.html#querystring_querystring_parse_str_sep_eq_options) can include arrays of strings if a user is messing around with the query string.

Reply:
@mgaeta I don't think we have to worry about users messing with query strings. If they are doing that, then all bets are off. I think this warrants a comment saying this only happens if the user is messing with query strings and wouldn't happen normally.